core/src/main/java/com/linecorp/armeria/client/endpoint/EndpointSelectionStrategy.java

```diff
@@ -34,9 +38,19 @@
      */
     EndpointSelectionStrategy WEIGHTED_ROUND_ROBIN = new WeightedRoundRobinStrategy();
 
+    /**
+     * Weighted round-robin strategy that increases weight gradually.
+     */
+    Function<EndpointWeighter, EndpointSelectionStrategy> SLOW_START_AWARE_WEIGHTED_ROUND_ROBIN =
+            SlowStartAwareEndpointSelectionStrategy::new;
+
     /**
      * Creates a new {@link EndpointSelector} that selects an {@link Endpoint} from the specified
      * {@link EndpointGroup}.
      */
     EndpointSelector newSelector(EndpointGroup endpointGroup);
+
+    @FunctionalInterface
+    interface EndpointWeighter extends BiFunction<Endpoint, Duration, Integer> {
```

Comment:
We could use `long` and `int` instead of `Duration` and `Integer` to reduce GC pressure during slow-start.